File: samples/python/get_ticker.py
import os
import requests
from dotenv import load_dotenv

load_dotenv()

GEMINI_BASE_URL = os.getenv('GEMINI_BASE_URL', 'https://api.gemini.com/v1')

def get_ticker(symbol: str):
    """Fetch ticker data for a given symbol."""
    # Symbol should be lowercase for Gemini API
    symbol_lower = symbol.lower()
    url = f"{GEMINI_BASE_URL}/pubticker/{symbol_lower}"

    try:
        response = requests.get(url)
        response.raise_for_status()
        print('Ticker data:', response.json())
    except requests.exceptions.RequestException as e:
        print(f'Error fetching ticker: {e}')

if __name__ == '__main__':
    import sys

    # Show help message
    if '--help' in sys.argv or '-h' in sys.argv:
        print('Usage: python3 get_ticker.py [symbol]')
        print('Example: python3 get_ticker.py ethusd')
        print('Default symbol: btcusd')
        sys.exit(0)

    # Get symbol from command line or use default
    symbol = sys.argv[1] if len(sys.argv) > 1 else 'btcusd'
    get_ticker(symbol)

File: samples/typescript/src/getTicker.ts
import axios from 'axios';
import * as dotenv from 'dotenv';

dotenv.config();

const GEMINI_BASE_URL = process.env.GEMINI_BASE_URL || 'https://api.gemini.com/v1';

async function getTicker(symbol: string): Promise<void> {
  try {
    // Symbol should be lowercase for Gemini API
    const symbolLower = symbol.toLowerCase();
    const response = await axios.get(`${GEMINI_BASE_URL}/pubticker/${symbolLower}`);

    console.log('Ticker data:', response.data);
  } catch (error) {
    console.error('Error fetching ticker:', error);
  }
}

// Get symbol from command line or use default
const symbol = process.argv[2] || 'btcusd';

if (process.argv.includes('--help') || process.argv.includes('-h')) {
  console.log('Usage: npx ts-node src/getTicker.ts [symbol]');
  console.log('Example: npx ts-node src/getTicker.ts ethusd');
  console.log('Default symbol: btcusd');
  process.exit(0);
}

getTicker(symbol);
